### Задание 1

Даны значения величины заработной платы заемщиков банка (salary) и значения их поведенческого кредитного скоринга (scoring):

salary = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]
scoring = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]

1. Найдите ковариацию этих двух величин: сначала без использования специальных функций, а затем с помощью функции numpy.cov. Полученные значения должны быть равны.
2. Найдите коэффициент корреляции Пирсона: с помощью ковариации и средних квадратических отклонений двух признаков, а затем с помощью специальных функций из библиотек numpy и pandas.

In [1]:
import numpy as np

In [2]:
salary = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
scoring = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

print(salary, scoring)

[ 35  45 190 200  40  70  54 150 120 110] [401 574 874 919 459 739 653 902 746 832]


#### 1.

Считаем ковариацию без специальных функций:

In [3]:
M_salary = salary.mean()
M_scoring = scoring.mean()

cov = ((salary - M_salary) * (scoring - M_scoring)).sum() / (len(salary) - 1)
cov

10175.37777777778

Считаем ковариацию с помощью функции numpy.cov:

In [4]:
np.cov(salary, scoring)

array([[ 3882.93333333, 10175.37777778],
       [10175.37777778, 33854.32222222]])

Значение попарных ковариаций лежит вне диагонали и равно 10175.37777778, что совпадает с полученным при расчете без специальной функции.

#### 2.

Считаем коэффициент корреляции Пирсона с помощью ковариации и средних квадратических отклонений двух признаков:

In [5]:
corr = cov / (salary.std(ddof = 1) * scoring.std(ddof = 1))
corr

0.8874900920739164

Считаем коэффициент корреляции Пирсона с помощью функции numpy.corrcoef:

In [6]:
np.corrcoef(salary, scoring)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

Значение попарных корреляций лежит вне диагонали и равно 0.88749009, что совпадает с полученным при расчете без специальной функции.

### Задание 2

Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:

131, 125, 115, 122, 131, 115, 107, 99, 125, 111

Известно, что в генеральной совокупности IQ распределен нормально. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Из условия задачи дано:

- генеральная совокупность IQ распределена нормально
- значения IQ выборки студентов, обучающихся в местных технических вузах: 131, 125, 115, 122, 131, 115, 107, 99, 125, 111
- $\sigma$ - неизвестна
- $p = 0.95$
- $\alpha = 1 - p = 0.05$

Так как рапределение нормальное и дисперсия неизвестна, то для расчета доверительного интервала будем применять $t$ статистику:

In [7]:
from scipy import stats

In [8]:
IQ = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

n = len(IQ)
mean = IQ.mean()
std = IQ.std(ddof=1)

n, mean, std

(10, 118.1, 10.54566788359614)

Расчитываем квантили:

In [9]:
alpha = 0.05

t1 = stats.t.ppf(alpha / 2, df=n - 1)
t2 = stats.t.ppf(1 - alpha / 2, df=n - 1)

t1, t2

(-2.262157162740992, 2.2621571627409915)

Доверительный интервал для математического ожидания генеральной совокупности IQ с надежностью 0.95:

In [10]:
(mean + t1 * std / np.sqrt(n), mean + t2 * std / np.sqrt(n))

(110.55608365158724, 125.64391634841274)

### Задание 3

Известно, что рост футболистов в сборной распределен нормально с известной дисперсией 25. На выборке объёма 27 выборочное среднее составило 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Из условия задачи дано:

- рост футболистов в сборной распределен нормально
- $D(X) = 25$
- $p = 0.95$
- $\alpha = 1 - p = 0.05$
- $\overline{X} = 174.2$
- $n = 27$

Так как рапределение нормальное и дисперсия известна, то для расчета доверительного интервала будем применять $Z$ статистику:

In [11]:
mean = 174.2
sigma = np.sqrt(25)
n = 27
alpha = 0.05

Расчитываем квантили:

In [12]:
q1 = stats.norm.ppf(alpha / 2)
q2 = stats.norm.ppf(1 - alpha / 2)

q1, q2

(-1.9599639845400545, 1.959963984540054)

Доверительный интервал для математического ожидания роста футболистов с надежностью 0.95:

In [13]:
(mean + q1 * (sigma / (n ** 0.5)), mean + q2 * (sigma / (n ** 0.5)))

(172.3140237765397, 176.08597622346028)